nodejs/sqlcommenter-nodejs/packages/sqlcommenter-sequelize/index.js

```diff
@@ -12,29 +12,44 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-const {hasComment, toW3CTraceContext} = require('./util');
-const {tracer} = require('@opencensus/nodejs');
+const {hasComment} = require('./util');
+const {OpenCensusProvider, OpenTelemetryProvider} = require('./provider')
 
 const defaultFields = {
     'route': true,
     'tracestate': false,
     'traceparent': false,
 };
 
+const providerMap = {
+    'OpenCensus': new OpenCensusProvider(),
+    'OpenTelemetry': new OpenTelemetryProvider(),
+}
+
 /**
  * All available variables for the commenter are on the `util.fields` object
  * passing the excludes parameter will result in each item being excluded from
  * the commenter output
  * 
  * @param {Object} sequelize 
- * @param {Object} includes a map of values to be optionally included.
+ * @param {Object} include - A map of values to be optionally included.
+ * @param {Object} options - A configuration object specifying where to collect trace data from. Accepted fields are:
+ *  TraceProvider: Should be either 'OpenCensus' or 'OpenTelemetry', indicating which library to collect trace data from.
  * @return {void}
  */
-exports.wrapSequelize = (sequelize, include={}) => {
+exports.wrapSequelize = (sequelize, include={}, options={TraceProvider: 'OpenCensus'}) => {
 
     /* c8 ignore next 2 */
     if (sequelize.___alreadySQLCommenterWrapped___)
         return;
+    // Resolve trace provider
+    if (!options.TraceProvider) {
+        options.TraceProvider = 'OpenCensus'; // Default to OpenCensus
+    }
+    let traceProvider = providerMap[options.TraceProvider];
+    if (!traceProvider) {
+        traceProvider = OpenCensusProvider;
```

Comment:
I fixed this and reworked it in my update to be clearer. And as for disabling tracing, I think the expected behavior of old clients is to keep appending trace data through OpenCensus, so that option should be non-default. Any thoughts @odeke-em ?